server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -2382,74 +2372,6 @@ public boolean pendingInSync() {
 
     public void maybeCheckIndex() {
         recoveryState.setStage(RecoveryState.Stage.VERIFY_INDEX);
```

Comment:
I think we can remove this method entirely?  As it's not actually running CheckIndex anymore.

Reply:
I thought so too - but the recovery stage is expected in some tests. That's an enum that's also streamed - I didn't want to mess with that too.

Reply:
Ah right, `setStage` actually does stuff rather than just updating a state variable, although it looks as though it's still just updating a state machine and starting/stopping some timers.  Maybe replace the calls to `maybeCheckIndex()` directly with `recoveryState.setStage(RecoveryState.Stage.VERIFY_INDEX);` then?  The important thing is that it isn't running CheckIndex so the method name is misleading now.

Reply:
Inlined the calls.